Update Security Compatibility with MySQL (#5921)
pingcap/docs: system-variables.md

- This variable indicates the location where data is stored. This location can be a local path or point to a PD server if the data is stored on TiKV.
- A value in the format of `ip_address:port` indicates the PD server that TiDB connects to on startup.

### default_authentication_plugin

- Scope: GLOBAL
- Default value: `mysql_native_password`

> - This variable sets the authentication method the server advertises during connection establishment.

Reviewer: @dveeden is this more correct?:
```suggestion
- This variable sets the default authentication method for users connecting to TiDB.
- Changes to this variable will only affect new users. To modify the authentication plugin of an existing user, use the [`ALTER USER`](/sql-statements/sql-statement-alter-user.md) statement.
```

Author: No that's not correct.

This sets the authentication method that is advertised in the server greeting. This may or may not be used by clients to then send a 'login' packet with that authentication method.

The MySQL 8.0 client by default ignores this and uses `caching_sha2_password` unless there is a explicit method set in the client (unlikely). See https://bugs.mysql.com/bug.php?id=93044 for details. 

If the authentication method send by the client matches that of the actual useraccount (`mysql.user.plugin`) then the authentication will proceed and if not the server send an AuthSwitchRequest to the client with info on what plugin should be used for that specific account.

In MySQL 8.0 this variable also influences the plugin that is used for `CREATE USER` if no plugin is specified. That's not the case in TiDB.

Reviewer: OK thanks for clarifying. Lets disregard my comment, it LGTM.

Reviewer 2: ```suggestion
- This variable sets the authentication method that the server advertises when the server-client connection is being established.
```

> 

Reviewer 2: Are the value options of this variable documented in `## Authentication plugin status`?

Author: Yes. Updated the text to add a link

Author: Yes, only `mysql_native_password` and `caching_sha2_password` are allowed and TLS client authentication works differently: https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers#enable-authentication

- This variable indicates the location where data is stored. This location can be a local path or point to a PD server if the data is stored on TiKV.
- A value in the format of `ip_address:port` indicates the PD server that TiDB connects to on startup.

### default_authentication_plugin

- Scope: GLOBAL
- Default value: `mysql_native_password`

> - This variable sets the authentication method that the server advertises when the server-client connection is being established. Possible values for this variable are documented in [Authentication plugin status](/security-compatibility-with-mysql.md#authentication-plugin-status)

Reviewer 2: ```suggestion
- This variable sets the authentication method that the server advertises when the server-client connection is being established. Possible values for this variable are documented in [Authentication plugin status](/security-compatibility-with-mysql.md#authentication-plugin-status).
```

> - Value options: `mysql_native_password` and `caching_sha2_password`. For more details, see [Authentication plugin status](/security-compatibility-with-mysql.md#authentication-plugin-status).

Reviewer 2: Follow up: add "since v5.2" back once the release-5.2 branch is available.

> - Value options: `mysql_native_password` and `caching_sha2_password` (since v5.2.0). For more details, see [Authentication plugin status](/security-compatibility-with-mysql.md#authentication-plugin-status).

Reviewer 2: ```suggestion
- Value options: `mysql_native_password` and `caching_sha2_password`. For more details, see [Authentication plugin status](/security-compatibility-with-mysql.md#authentication-plugin-status).
```
